crates/state/src/batch.rs

```diff
@@ -70,9 +70,9 @@ impl SignedBatchCheckpoint {
         &self.inner
     }
 
-    pub fn verify_sig(&self, pub_key: Buf32) -> bool {
+    pub fn verify_sig(&self, pub_key: &Buf32) -> bool {
         let msg = self.checkpoint().get_sighash();
-        verify_schnorr_sig(&self.signature, &msg, &pub_key)
+        verify_schnorr_sig(&self.signature, &msg, pub_key)
```

Comment:
Why do we need to take ownership of `pub_key` can't we verify it by reference?
A lot of copies here... might have some perf issues in the future

Reply:
Looks good now?

Reply:
Yeah